mautic/mautic
# Modified the way we are pushing leads to SF

… records found related to the lead created in Mautic and/or email Lead/Contact

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | (enhancement)
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR enhances the way we are pushing leads into Salesforce. First we search Mautic's database records to find the related SF id with the mautic contact Id. If it's not found then we go to SF to search for a lead created or converted that is related to an email account that we are pushing into SF. If any data is found in either DB's then we update all the records found. If none are found then we create a new lead in SF

#### Steps to test this PR:
1. Apply this PR, Use a form or a campaign to push a contact to Salesforce Data must contain an email address or have been stored in Mautic's database prior.

2. Notice that we are fetching the id's stored in the integration_entity where the internal object is Lead and related to the created lead in Mautic.  

2. Secondly if we did not find the records in Mautic's db then we go into SF to look for that id, and if that lead was converted then we update the contact record instead.

3. Convert a record from lead to contact in SF and notice if the contact is updated with the pushed data.

4. If it's a new email address then a new Lead in Salesforce should be created
